zwave_js_server/model/node/firmware.py

```diff
@@ -133,193 +137,125 @@ class NodeFirmwareUpdateStatus(IntEnum):
     OK_RESTART_PENDING = 255
 
 
-class NodeFirmwareUpdateProgressDataType(TypedDict):
+class NodeFirmwareUpdateProgressDataType(FirmwareUpdateProgressDataType):
     """Represent a node firmware update progress dict type."""
 
     currentFile: int
     totalFiles: int
-    sentFragments: int
-    totalFragments: int
-    progress: float
 
 
 @dataclass
-class NodeFirmwareUpdateProgress:
+class NodeFirmwareUpdateProgress(FirmwareUpdateProgress):
     """Model for a node firmware update progress data."""
 
     node: Node
     data: NodeFirmwareUpdateProgressDataType = field(repr=False)
     current_file: int = field(init=False)
     total_files: int = field(init=False)
```

Comment:
Do you mean the attributes `current_file` and `total_files`?

If it's not supported now, I don't think we should add it. There's no problem with adding them later when they are supported by driver updates.